JavaScript/JavaScriptSDK/SendBuffer.ts

```diff
@@ -183,6 +187,16 @@ module Microsoft.ApplicationInsights {
             var sentElements = this.getBuffer(SessionStorageSendBuffer.SENT_BUFFER_KEY);
             sentElements = sentElements.concat(payload);
 
+            if (sentElements.length > SessionStorageSendBuffer.MAX_BUFFER_SIZE) {
```

Comment:
Ideally we should check the total buffer size (BUFFER + SENT_BUFFER) in `enqueue` method, but I try to minimize the number of times we read from session storage. This is why `enqueue` method only checks the BUFFER size. IMHO it's OK to check the size in `markAsSent` method because this is one of two places where we add new items to the SENT_BUFFER (the other one is in the constructor).